e2e/nomostest/config_sync.go

```diff
@@ -87,8 +89,59 @@ var (
 	reconcilerPollingPeriod time.Duration
 	// hydrationPollingPeriod specifies hydration-controller polling period as time.Duration
 	hydrationPollingPeriod time.Duration
+
+	// DefaultRootReconcilerName is the root-reconciler name of the default RootSync object: "root-sync".
+	DefaultRootReconcilerName = core.RootReconcilerName(configsync.RootSyncName)
+	// DefaultRootRepoNamespacedName is the NamespacedName of the default RootSync object.
+	DefaultRootRepoNamespacedName = RootSyncNN(configsync.RootSyncName)
+	// DefaultRootSyncID is the ID of the default RootSync object.
+	DefaultRootSyncID = RootSyncID(configsync.RootSyncName)
 )
 
+// RootSyncNN returns the NamespacedName of the RootSync object.
+func RootSyncNN(name string) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: configsync.ControllerNamespace,
+		Name:      name,
+	}
+}
+
+// RepoSyncNN returns the NamespacedName of the RepoSync object.
+func RepoSyncNN(ns, name string) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: ns,
+		Name:      name,
+	}
+}
+
+// RootSyncID returns an ID for the specified RootSync object.
+func RootSyncID(name string) core.ID {
+	return core.ID{
+		GroupKind: schema.GroupKind{
+			Group: v1beta1.SchemeGroupVersion.Group,
+			Kind:  configsync.RootSyncKind,
+		},
+		ObjectKey: client.ObjectKey{
+			Name:      name,
+			Namespace: configsync.ControllerNamespace,
+		},
+	}
+}
+
+// RepoSyncID returns an ID for the specified RepoSync object.
+func RepoSyncID(name, namespace string) core.ID {
```

Comment:
nit: types.NamespacedName is a bit more user friendly as a parameter. No need to remember the argument ordering of name vs namespaces

Reply:
While true, it would actually require a fair amount of refactoring to ensure all the callers have an NN available. In this refactor, most of the uses have name and namespace as some combination of constants and hard-coded strings. Then the NN is derived from the id.ObjectKey, rather than the other way around.

I agree NN is better on usability from an ordering perspective, but if it turns all the one-liners into two-liners, I'm not sure it's all that much better. 

That said, a lot of the existing functions take name and namespace args, and they're not always in the same order. Converting them all to NN might make sense as a follow up PR, if you're interested.

Reply:
Yeah the main thing that bothers me is the inconsistent ordering, which can be a bit of a gotcha. It would be a pretty big refactor to replace with NamespacedName across the board

Reply:
I hear you. Mostly I let the IDE tell me what the order is, which makes it less of an issue. But if you think you can make it better, go for it.